src/Microsoft.ComponentDetection.Common/DependencyGraph/ComponentRecorder.cs

```diff
@@ -62,6 +62,24 @@ public IEnumerable<DetectedComponent> GetDetectedComponents()
         return detectedComponents;
     }
 
+    public IEnumerable<string> GetSkippedComponents()
+    {
+        IEnumerable<string> skippedComponents;
+        if (this.singleFileRecorders == null)
+        {
+            return Enumerable.Empty<string>();
+        }
+
+        skippedComponents = this.singleFileRecorders
```

Comment:
You can simpify this query: 

```csharp
this.singleFileRecorders.Select(x => x.GetSkippedComponents().Keys).SelectMany(x => x).Distinct().ToImmutableList();
```

Reply:
You can also return directly here. No need to split declaration, initialization, and return statement.

Reply:
Done.